scripts/add_new_provider.py

```diff
@@ -10,33 +10,40 @@
 from django.core.validators import URLValidator
 import libcloud.security
 
-from urlparse import urlparse
+from urlparse import urlparse, urljoin
 
 from core.models import Provider, PlatformType, ProviderType, Identity, Group,\
     IdentityMembership, AccountProvider, Quota, ProviderInstanceAction
 from core.models import InstanceAction
+from service.driver import get_account_driver
+from atmosphere.settings import secrets
+from atmosphere import settings
 
 libcloud.security.VERIFY_SSL_CERT = False
 libcloud.security.VERIFY_SSL_CERT_STRICT = False
 KVM = PlatformType.objects.get_or_create(name='KVM')[0]
 XEN = PlatformType.objects.get_or_create(name='Xen')[0]
 openstack = ProviderType.objects.get_or_create(name='OpenStack')[0]
 
-valid_url = URLValidator()
+url_validator = URLValidator()
 
 
-def require_input(question, validate_answer=None):
+def require_input(
```

Comment:
I think we could get by with just a small tweak to the original interface:

`require_input(input, default=None, validator=None)`. I cannot think of any edge cases that it wouldn't cover. Looking at the new interface it allows for conflicting behavior like providing a validator but setting use_validate_answer to false. I'd be willing to contribute a patch for this. Comments?